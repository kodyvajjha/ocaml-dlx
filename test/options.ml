let () =
  let items = [ "a"; "b"; "c"; "d"; "e"; "f"; "g" ] in
  let options =
    [
      [ "c"; "e" ];
      [ "a"; "d"; "g" ];
      [ "b"; "c"; "f" ];
      [ "a"; "d"; "f" ];
      [ "b"; "g" ];
      [ "d"; "e"; "g" ];
    ]
  in
  let cur = Dlx.Algox.mk ~items ~options in
  let opts =
    CCList.map (fun i -> Dlx.Algox.option_of i cur) [ 9; 12; 16; 20; 24; 27 ]
  in
  CCFormat.printf "@.%a" CCFormat.Dump.(list (list string)) opts
