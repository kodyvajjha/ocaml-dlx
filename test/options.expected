
[["c";"e"];["a";"d";"g"];["b";"c";"f"];["a";"d";"f"];["b";"g"];["d";"e";"g"]]